integration/sofa-rpc/src/main/java/io/seata/integration/sofa/rpc/TransactionContextProviderFilter.java

```diff
@@ -55,8 +55,8 @@ public SofaResponse invoke(FilterInvoker filterInvoker, SofaRequest sofaRequest)
         }
         boolean bind = false;
         if (xid != null) {
-            RpcInternalContext.getContext().setAttachment(RootContext.KEY_XID, xid);
-            RpcInternalContext.getContext().setAttachment(RootContext.KEY_BRANCH_TYPE, branchType.name());
+            RpcInternalContext.getContext().setAttachment(RootContext.HIDDEN_KEY_XID, xid);
+            RpcInternalContext.getContext().setAttachment(RootContext.HIDDEN_KEY_BRANCH_TYPE, branchType.name());
```

Comment:
Here are some case to prove   `RpcInternalContext.getContext().setAttachment(RootContext.KEY_XID, xid);` not as expected ?

Reply:
Test case added